Docs/source/developers/local_compile.rst

```diff
@@ -0,0 +1,136 @@
+.. _developers-local-compile:
+
+Fast, Local Compilation
+=======================
+
+For simplicity, WarpX :ref:`compilation with CMake <building-cmake>` by default downloads, configures and compiles compatible versions of :ref:`central dependencies <install-dependencies>` such as:
+
+* `AMReX <https://amrex-codes.github.io>`__
+* `PICSAR <https://github.com/ECP-WarpX/picsar>`__
+* `openPMD-api <https://github.com/openPMD/openPMD-api>`__
+* `pyAMReX <https://github.com/AMReX-Codes/pyamrex>`__
+* `pybind11 <https://github.com/pybind/pybind11>`__
+
+on-the-fly, which is called a *superbuild*.
+
+In some scenarios, e.g., when compiling without internet, with slow internet access, or when working on WarpX and its dependencies, modifications to the superbuild strategy might be preferable.
+In the below workflows, you as the developer need to make sure to use compatible versions of the dependencies you provide.
+
+
+.. _developers-local-compile-src:
+
+Compiling From Local Sources
+----------------------------
+
+This workflow is best for developers that make changes to WarpX, AMReX, PICSAR, openPMD-api and/or pyAMReX at the same time.
+For instance, use this if you add a feature in AMReX and want to try it in WarpX before it is proposed as a pull request for inclusion in AMReX.
+
+Instead of downloading the source code of the above dependencies, one can also use an already cloned source copy.
+For instance, clone these dependencies to ``$HOME/src``:
+
+.. code-block:: bash
+
+   cd $HOME/src
+
+   git clone https://github.com/ECP-WarpX/WarpX.git warpx
+   git clone https://github.com/AMReX-Codes/amrex.git
+   git clone https://github.com/openPMD/openPMD-api.git
+   git clone https://github.com/ECP-WarpX/picsar.git
+   git clone https://github.com/AMReX-Codes/pyamrex.git
+   git clone https://github.com/pybind/pybind11.git
+
+Now modify the dependencies as needed in their source locations, update sources if you cloned them earlier, etc.
+When building WarpX, :ref:`the following CMake flags <building-cmake-options>` will use the respective local sources:
+
+.. code-block:: bash
+
+   cd src/warpx
+
+   rm -rf build
+
+   cmake -S . -B build  \
+     -DWarpX_PYTHON=ON  \
```

Comment:
This could be potentially confusing since the Python option is not needed because of local compilation. Also, the build comment below is not consistent since it doesn't have the `--target pip_install` option. Maybe leave out the Python related options for simplicity in this example?

Reply:
Good catch, adding `--target pip_install`.

I would show with Python, since we want to show this by default. Also, that enables me to show how to use a local pybind11 and pyamrex.

Developers understand that they can turn it off when they read this part.

Reply:
Target added :)